nsxt/resource_nsxt_policy_segment_security_profile.go

```diff
@@ -0,0 +1,317 @@
+/* Copyright © 2020 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	gm_infra "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/global_infra"
+	gm_model "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtPolicySegmentSecurityProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicySegmentSecurityProfileCreate,
+		Read:   resourceNsxtPolicySegmentSecurityProfileRead,
+		Update: resourceNsxtPolicySegmentSecurityProfileUpdate,
+		Delete: resourceNsxtPolicySegmentSecurityProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"bpdu_filter_allow": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.IsMACAddress,
+				},
+				Optional: true,
+			},
+			"bpdu_filter_enable": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"dhcp_client_block_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"dhcp_client_block_v6_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"dhcp_server_block_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"dhcp_server_block_v6_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"non_ip_traffic_block_enabled": {
```

Comment:
Aren't we usually adding description to resource attributes?

Reply:
This file was generated by policygen, and descriptions in spec are often too long to appear here in the code. We do have them in documentation. I'm not sure description in the code has any practical purpose.

Reply:
Makes sense. I wonder how much the provider image will shrink if we remove all these strings :)